frontend/src/app/services/auth-services/auth.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { map, tap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+type AuthResponse = {
+  token: string;
+  user: any;
+};
+export type UserDto = {
+  id?: string;
+  username: string;
+  email: string;
+};
+
+@Injectable({ providedIn: 'root' })
+export class AuthService {
+  private http = inject(HttpClient);
+  private api = 'http://localhost:8080';
+
+  login(body: { email: string; password: string }) {
+    return new Promise<AuthResponse>((resolve) =>
+      setTimeout(() => resolve({ token: 'temp-token', user: { email: body.email } }), 400),
+    ) as any;
+  }
+
+  signup(body: { username: string; email: string }) {
+    return this.http.post<UserDto>(`${this.api}/users`, body).pipe(
+      map((user) => ({ token: 'temp-token', user }) as AuthResponse),
+      tap((res) => localStorage.setItem('token', res.token)),
+    );
+  }
+
+  getUsers(): Observable<UserDto[]> {
+    return this.http.get<UserDto[]>(`${this.api}/users`);
+  }
```

Comment:
prefer promise here, over the observable, since this use case is quite simple to introduce this much complexity in the application
```suggestion
  getUsers(): Promise<UserDto[]> {
    return firstValueFrom(this.http.get<UserDto[]>(`${this.api}/users`));
  }
```
and then use await when you call the method